lib/ansible/modules/cloud/amazon/rds_v2.py

```diff
@@ -0,0 +1,1148 @@
+
+    try:
+        response = client.describe_db_instances(**params)
+
+    except botocore.exceptions.ClientError as e:
+        if 'DBInstanceNotFound' not in e.message:
+            module.fail_json(msg=str(e))
+        else:
+            response = None
+
+    return response
+
+
+def get_db_snapshot(client, module, snapshot_name=None):
+    params = dict()
+    if module.params.get('db_instance_identifier'):
+        params['DBInstanceIdentifier'] = module.params.get('db_instance_identifier')
+    if snapshot_name:
+        params['DBSnapshotIdentifier'] = snapshot_name
+    else:
+        params['DBSnapshotIdentifier'] = module.params.get('db_snapshot_identifier')
+
+    try:
+        response = client.describe_db_snapshots(**params)
+    except botocore.exceptions.ClientError as e:
+        if 'DBSnapshotNotFound' not in e.message:
+            module.fail_json(msg=str(e))
+        else:
+            response = None
+
+    return response
+
+
+def create_db_instance(client, module):
+    # DBClusterIdentifier is only for Aurora instances. If one is supplied, only a subset of params are supported.
+    if module.params.get('db_cluster_identifier'): 
+        required_vars = ['db_instance_identifier','db_instance_class', 'engine']
+        valid_vars = ['db_cluster_identifier', 'availability_zone', 'db_subnet_group_name',
+                  'preferred_maintenance_window', 'db_parameter_group_name',
+                  'engine_version', 'auto_minor_version_upgrade', 'license_model', 'option_group_name',
+                  'publicly_accessible', 'character_set_name', 'storage_encrypted', 'kms_key_id', 'tags']
+    else:
+        required_vars = ['db_instance_identifier', 'engine', 'db_instance_class', 'master_username', 'master_user_password']
+        valid_vars = ['db_name', 'allocated_storage', 'vpc_security_group_ids', 'availability_zone', 'db_subnet_group_name',
+                  'preferred_maintenance_window', 'db_parameter_group_name', 'backup_retention_period', 'preferred_backup_window',
+                  'port', 'multi_az', 'engine_version', 'auto_minor_version_upgrade', 'license_model', 'iops',
+                  'option_group_name', 'publicly_accessible', 'character_set_name', 'storage_encrypted', 'kms_key_id', 'tags',
+                  'db_cluster_identifier', 'copy_tags_to_snapshot', 'monitoring_interval', 'storage_type']
+
+    params = validate_parameters(required_vars, valid_vars, module)
+    # Check if the db instance already exists
+    db_instance = get_db_instance(client, module)
+    if db_instance:
+        response = db_instance
+        changed=False
+    else:
+        try:
+            response = client.create_db_instance(**params)
+            changed = True
+        except botocore.exceptions.ClientError as e:
+            e = get_exception()
+            module.fail_json(msg=str(e))
+
+        if module.params.get('wait'):
+            status_achieved, response = wait_for_status(client, module, 'available')
```

Comment:
There's a new AWS Retry generic handler that you can use here.  Check out the ec2 shared code in module_utils